src/neuroconv/datainterfaces/behavior/neuralynx/neuralynx_nvt_interface.py

```diff
@@ -0,0 +1,142 @@
+        metadata_schema = super().get_metadata_schema()
+
+        if "Behavior" not in metadata_schema["properties"]:
+            metadata_schema["properties"]["Behavior"] = get_base_schema(required=[self.nvt_filename])
+            metadata_schema.setdefault("requirements", []).append("Behavior")
+        else:
+            metadata_schema["properties"]["Behavior"].setdefault("requirements", []).append(self.nvt_filename)
+
+        metadata_schema["properties"]["Behavior"]["properties"].update(
+            {
+                self.nvt_filename: dict(
+                    type="object",
+                    required=["name", "reference_frame"],
+                    properties=dict(
+                        name=dict(type="string", default="NvtSpatialSeries"),
+                        reference_frame=dict(type="string", default="unknown"),
+                    ),
+                )
+            }
+        )
+
+        return metadata_schema
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        add_position: bool = True,
+        add_angle: Optional[bool] = None,
+    ):
+        """
+        Add NVT data to a given in-memory NWB file
+
+        Parameters
+        ----------
+        nwbfile : NWBFile
+            nwb file to which the recording information is to be added
+        metadata : dict, optional
+            metadata info for constructing the nwb file.
+        add_position : bool, default=True
+        add_angle : bool, optional
+            If None, write angle as long as it is not all 0s
+        """
+
+        metadata = (metadata or self.get_metadata()).to_dict()
+
+        data = read_data(self.file_path)
+
+        if add_position:
+            # convert to float and change <= 0 (null) to NaN
+            xi = data["Xloc"]
+            x = xi.astype(float)
+            x[xi <= 0] = np.nan
+
+            yi = data["Yloc"]
+            y = yi.astype(float)
+            y[yi <= 0] = np.nan
+
+            spatial_series = SpatialSeries(
+                name=metadata["Behavior"][self.nvt_filename]["name"],
+                data=np.c_[x, y],
+                reference_frame=metadata["Behavior"][self.nvt_filename]["reference_frame"],
+                unit="pixels",
+                conversion=1.0,
+                timestamps=self.get_timestamps(),
+                description=f"Pixel x and y coordinates from the .nvt file with header data: {json.dumps(self.header, cls=NWBMetaDataEncoder)}",
+            )
+
+            nwbfile.add_acquisition(Position([spatial_series], name="NvtPosition"))
+
+        if add_angle or (add_angle is None and not np.all(data["Angle"] == 0)):
+            nwbfile.add_acquisition(
+                CompassDirection(
+                    SpatialSeries(
+                        name="NvtAngleSpatialSeries",
+                        data=data["Angle"],
+                        reference_frame="unknown",
```

Comment:
Same reference frame would apply to the angular series as well

Reply:
Or would it? For the angular, might need to additionally specify what convention the incidence is with respect to separately from the main x/y reference frame?

Reply:
It would need to be different. Reference frame here would correspond to where 0 degrees is, which I assume with either be up, going clockwise or to the right going counter clockwise.